app/code/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/Edit.php

```diff
@@ -57,6 +59,12 @@ public function execute()
                 $this->_redirect('sales_rule/*');
                 return;
             }
+
+            // set entered data if was error when we do save
+            $data = $this->_objectManager->get(\Magento\Backend\Model\Session::class)->getPageData(true);
```

Comment:
If I'm not wrong, the `\Magento\Backend\Model\Session` is already in the `\Magento\Framework\App\Action\Context` which is an injected dependency, so there is no need to instantiate it through the object manager, what do you think?

Reply:
@aleron75 yes we can remove the objectManager here. i just moved the block to resolve this issue .